apps/wallet/src/ui/app/pages/home/receipt/index.tsx

```diff
@@ -19,27 +16,16 @@ import st from './ReceiptPage.module.scss';
 function ReceiptPage() {
     const [searchParams] = useSearchParams();
     const [showModal, setShowModal] = useState(true);
-    const dispatch = useAppDispatch();
     // get tx results from url params
     const txDigest = searchParams.get('txdigest');
 
     const transferType = searchParams.get('transfer') as 'nft' | 'coin';
 
-    const txResults: TxResultState[] = useAppSelector(
-        ({ txresults }) => txresults.latestTx
-    );
-
-    const loading: boolean = useAppSelector(
-        ({ txresults }) => txresults.loading
-    );
-
-    useEffect(() => {
-        dispatch(getTransactionsByAddress());
-    }, [dispatch]);
+    const { data, isLoading } = useRecentTransactions();
```

Comment:
would it be better to use a query that fetches only the requested tx?

Reply:
Yes, but I'll leave that for #7437 to resolve. This is just supposed to be a quick fix to improve this while we land other improvements.